pkg/volume/glusterfs/glusterfs.go

```diff
@@ -160,8 +158,33 @@ func (plugin *glusterfsPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volu
 	return plugin.newMounterInternal(spec, ep, pod, plugin.host.GetMounter(plugin.GetPluginName()))
 }
 
+func (plugin *glusterfsPlugin) getEndpointNameAndNamespace(spec *volume.Spec, defaultNamespace string) (string, string, error) {
+	if spec.Volume != nil && spec.Volume.Glusterfs != nil {
+		endpoints := spec.Volume.Glusterfs.EndpointsName
+		if endpoints != "" {
```

Comment:
either return `"", defaultNamespace, nil` here or a "no glusterFS endpoint specified" error, falling through to the "Spec does not reference a GlusterFS volume type" error is not correct

Reply:
Done